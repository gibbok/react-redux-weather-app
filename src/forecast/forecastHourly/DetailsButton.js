// @flow
import * as React from 'react'
import {Button} from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +onDetailsClick: Function,
}
/* eslint-enable no-undef */

const DetailsButton:React.StatelessFunctionalComponent<any> = ({ onDetailsClick }:PropsType):React.Element<any> => {
  return (
    <Button onClick={(e) => onDetailsClick(e)}>Details</Button>
  )
}

export default DetailsButton
