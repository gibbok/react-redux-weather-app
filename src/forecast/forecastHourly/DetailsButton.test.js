import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import sinon from 'sinon'
import DetailsButton from './DetailsButton'

describe('<DetailsButton />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <DetailsButton onClick={() => {}} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should simulate click event', () => {
    const onDetailsClick = sinon.spy()
    let wrapper = shallow(<DetailsButton onDetailsClick={onDetailsClick} />)
    wrapper.find('button').simulate('click')
    expect(onDetailsClick.called).toEqual(true)
  })
})
