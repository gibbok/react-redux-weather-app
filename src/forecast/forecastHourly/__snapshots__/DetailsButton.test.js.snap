// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsButton /> should render 1`] = `
<withStyles(Button)
  onClick={[Function]}
  raised={true}
>
  Details
</withStyles(Button)>
`;
